src/text-editor.coffee

```diff
@@ -677,7 +677,7 @@ class TextEditor extends Model
   # this editor.
   shouldPromptToSave: ({windowCloseRequested}={}) ->
     if windowCloseRequested
-      @isModified()
+      false
```

Comment:
It will be tied to the workspace state, which is keyed on the project paths. Making it independent of the workspace state is a substantial increase in scope.